tests/st/calicoctl/test_default_pools.py

```diff
@@ -0,0 +1,180 @@
+    def tearDownClass(cls):
+        for host in cls.hosts:
+            host.cleanup()
+
+    @parameterized.expand([
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/27",  None),    # Too small
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/32",  None),    # Too small, but legal
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/33",  None),    # Impossible CIDR
+        (False, "CALICO_IPV4POOL_CIDR", "256.0.0.0/24", None),    # Invalid IP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24",  None),    # Typical non-default pool
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/26",  None),    # Smallest legal pool
+        (True,  "CALICO_IPV6POOL_CIDR", "fd00::/122",   None),    # Smallest legal pool
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/123",   None),    # Too small
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/128",   None),    # Too small, but legal
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/129",   None),    # Impossible CIDR
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "cross-subnet"),  # Typ. non-def pool, IPIP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "always"),  # Typical non-default pool, IPIP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "off"),  # Typical pool, explicitly no IPIP
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", "always"),  # IPv6 - IPIP not permitted
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", "cross-subnet"),  # IPv6 - IPIP not permitted
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", "off"),      # IPv6, IPIP explicitly off
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/122", "junk"),     # Invalid IPIP value
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "reboot"),  # Invalid IPIP value
+        (True,  "CALICO_IPV4POOL_CIDR", "0.0.0.0/0",    None),    # Valid, but not sensible
+        (True,  "CALICO_IPV6POOL_CIDR", "::/0",         None),    # Valid, but not sensible
+        (True,  "CALICO_IPV6POOL_CIDR", "fd80::0:0/120", None),   # Valid, but non-canonical form
+        (False, "CALICO_IPV6POOL_CIDR", "1.2.3.4/24",    None),   # Wrong type
+        (False, "CALICO_IPV4POOL_CIDR", "fd00::/24",     None),   # Wrong type
+        (True,  "CALICO_IPV6POOL_CIDR", "::0:a:b:c:d:e:0/120", None),  # Valid, non-canonical form
+        (False, "CALICO_IPV4POOL_CIDR", "1.2/16",       None),    # Valid, unusual form
+    ])
+    def test_default_pools(self, success_expected, param, value, ipip):
+        """
+        Test that the various options for default pools work correctly
+        """
+        # Get command line for starting docker
+        output = self.host.calicoctl("node run --dryrun")
+        base_command = output.split('\n')[-4].rstrip()
+
+        # Modify command line to add the options we want to test
+        # Command line will attempt to use quay.io/calico/node:latest - we want calico/node:latest
+        base_command = base_command.replace("quay.io/", "")
+        env_index = base_command.find("-e")
```

Comment:
This might be a bit cleaner since it avoids the indexing, which is always fiddly:
```
env_inserts = "-e %s=%s " % (param, value)
if ipip is not None:
    env_inserts += "-e CALICO_IPV4POOL_IPIP=%s " % ipip
prefix, _, suffix := base_command.partition("-e")
command = prefix + env_inserts + "-e" + suffix
```